packages/gatsby-plugin-page-creator/src/get-collection-route-params.ts

```diff
@@ -1,29 +1,34 @@
+import { extractFieldWithoutUnion, removeFileExtension } from "./path-utils"
+
 // This extracts params from its filePath counerpart
 // and returns an object of it's matches.
-// e.g.,
-
-import {
-  compose,
-  extractFieldWithoutUnion,
-  removeFileExtension,
-} from "./path-utils"
-
-//   /foo/{Product.id}, /foo/123 => {id: 123}
+// e.g. /foo/{Product.id}, /foo/123 => {id: 123}
 export function getCollectionRouteParams(
   urlTemplate: string,
   urlPath: string
 ): Record<string, string> {
   const params = {}
-  // remove the starting path to simplify the loop
-  const urlTemplateParts = urlTemplate.split(`/`)
+
```

Comment:
As we now support prefixes/postfixes the strings between `/` don't only contain the relevant params but also other strings. A regex pattern is created out of the template to extract them anyways